api/src/main/java/org/openmrs/api/db/ConceptDAO.java

```diff
@@ -84,42 +83,74 @@
 	
 	/**
+	 * Get the concept name
+	 * 
 	 * @see org.openmrs.api.ConceptService#getConceptName(java.lang.Integer)
-	 * @param conceptNameId
-	 * @return The ConceptName matching the specified conceptNameId
+	 * @param conceptNameId The concept name id
+	 * @return The concept name matching the specified conceptNameId
+	 * @throws DAOException
 	 */
 	public ConceptName getConceptName(Integer conceptNameId) throws DAOException;
 	
 	/**
+	 * Returns all concepts
+	 * 
 	 * @see org.openmrs.api.ConceptService#getAllConcepts(java.lang.String, boolean, boolean)
+	 * @throws DAOException
 	 */
 	public List<Concept> getAllConcepts(String sortBy, boolean asc, boolean includeRetired) throws DAOException;
 	
 	/**
 	 * Returns a list of concepts based on the search criteria
 	 * 
-	 * @param name
-	 * @param loc
+	 * @param name The name of the concept to search
+	 * @param loc The locale to look in
 	 * @param searchOnPhrase This puts wildcard characters around the concept name search criteria
-	 * @return List&lt;Concept&gt;
+	 * @return a list of concepts based on search criteria
 	 * @throws DAOException
 	 * <strong>Should</strong> not return concepts with matching names that are voided
 	 */
 	public List<Concept> getConcepts(String name, Locale loc, boolean searchOnPhrase, List<ConceptClass> classes,
 	        List<ConceptDatatype> datatypes) throws DAOException;
 	
 	/**
+	 * Return a list of concept results based on search criteria
+	 * 
 	 * @see ConceptService#getConcepts(String, List, boolean, List, List, List, List, Concept,
 	 *      Integer, Integer)
+	 * @param phrase 
+	 * @param locales A list of locales
+	 * @param includeRetired 
+	 * @param requireClasses 
+	 * @param excludeClasses 
+	 * @param requireDatatypes 
+	 * @param excludeDatatypes 
+	 * @param answersToConcept 
+	 * @param start 
+	 * @param size 
+	 * @return a list of concept results
 	 * @throws DAOException
-	 * <strong>Should</strong> return correct results for concept with names that contains words with more weight
+	 * <strong>Should</strong> return correct results for concept with names that contain words with more weight
 	 * <strong>Should</strong> return correct results if a concept name contains same word more than once
 	 */
 	public List<ConceptSearchResult> getConcepts(String phrase, List<Locale> locales, boolean includeRetired,
 	        List<ConceptClass> requireClasses, List<ConceptClass> excludeClasses, List<ConceptDatatype> requireDatatypes,
 	        List<ConceptDatatype> excludeDatatypes, Concept answersToConcept, Integer start, Integer size)
 	        throws DAOException;
-	
+	/**
+	 * Returns count of concepts
+	 * 
+	 * @param phrase
+	 * @param locales 
+	 * @param includeRetired 
+	 * @param requireClasses 
+	 * @param excludeClasses 
+	 * @param requireDatatypes 
+	 * @param excludeDatatypes 
+	 * @param answersToConcept 
+	 * @return the concept count
+	 * @throws DAOException
```

Comment:
Shouldn't we have some descriptions for these parameters?

Reply:
Same applies for these param , I could have just written plain statements following their names , but i was not pretty sure of what they actually do . for example consider the  param ** excludeDatatypes** , i would write **The Data types to exclude** , do you think that would portray the whole meaning  ?